src/db/migrate.rs

```diff
@@ -0,0 +1,189 @@
+             CREATE TABLE authors (
+                 id SERIAL PRIMARY KEY,
+                 name VARCHAR(255),
+                 email VARCHAR(255),
+                 slug VARCHAR(255) UNIQUE NOT NULL
+             );
+             CREATE TABLE author_rels (
+                 rid INT REFERENCES releases(id),
+                 aid INT REFERENCES authors(id),
+                 UNIQUE(rid, aid)
+             );
+             CREATE TABLE keywords (
+                 id SERIAL PRIMARY KEY,
+                 name VARCHAR(255),
+                 slug VARCHAR(255) NOT NULL UNIQUE
+             );
+             CREATE TABLE keyword_rels (
+                 rid INT REFERENCES releases(id),
+                 kid INT REFERENCES keywords(id),
+                 UNIQUE(rid, kid)
+             );
+             CREATE TABLE owners (
+                 id SERIAL PRIMARY KEY,
+                 login VARCHAR(255) NOT NULL UNIQUE,
+                 avatar VARCHAR(255),
+                 name VARCHAR(255),
+                 email VARCHAR(255)
+             );
+             CREATE TABLE owner_rels (
+                 cid INT REFERENCES releases(id),
+                 oid INT REFERENCES owners(id),
+                 UNIQUE(cid, oid)
+             );
+             CREATE TABLE builds (
+                 id SERIAL,
+                 rid INT NOT NULL REFERENCES releases(id),
+                 rustc_version VARCHAR(100) NOT NULL,
+                 cratesfyi_version VARCHAR(100) NOT NULL,
+                 build_status BOOL NOT NULL,
+                 build_time TIMESTAMP NOT NULL DEFAULT CURRENT_TIMESTAMP,
+                 output TEXT
+             );
+             CREATE TABLE queue (
+                 id SERIAL,
+                 name VARCHAR(255),
+                 version VARCHAR(100),
+                 attempt INT DEFAULT 0,
+                 date_added TIMESTAMP NOT NULL DEFAULT CURRENT_TIMESTAMP,
+                 UNIQUE(name, version)
+             );
+             CREATE TABLE files (
+                 path VARCHAR(4096) NOT NULL PRIMARY KEY,
+                 mime VARCHAR(100) NOT NULL,
+                 date_added TIMESTAMP NOT NULL DEFAULT CURRENT_TIMESTAMP,
+                 date_updated TIMESTAMP NOT NULL DEFAULT CURRENT_TIMESTAMP,
+                 content BYTEA
+             );
+             CREATE TABLE config (
+                 name VARCHAR(100) NOT NULL PRIMARY KEY,
+                 value JSON NOT NULL
+             );
+             CREATE INDEX ON releases (release_time DESC);
+             CREATE INDEX content_idx ON crates USING gin(content);",
+            // downgrade query
+            "DROP TABLE authors, author_rels, keyword_rels, keywords, owner_rels,
+                        owners, releases, crates, builds, queue, files, config;"
+        ),
+    ];
+
+    for migration in migrations {
+        migrator.register(migration);
+    }
+
+    if let Some(version) = version {
+        if version > migrator.current_version()?.unwrap_or(0) {
```

Comment:
I'm worried whether this will attempt to recreate all the tables if this is used with an existing database (that uses the `config` table to track its version). Should we include a check to look for a `database_version` entry in the `config` table before assuming an empty database?